ui/src/api.ts

```diff
@@ -25,14 +25,8 @@ export async function getAllPapers() {
 }
 
 export async function papers(s2Ids: string[]) {
-  const data = await doGet(
-    axios.get("/api/v0/papers", {
-      params: {
-        id: s2Ids.join(",")
-      }
-    })
-  );
-  return (data || []) as Paper[];
+  var promises = collectPromise(s2Ids, 50);
```

Comment:
@jocelynzsun can you pull this out into a named constant, with a detailed commend describing why we needed to split up the list of papers for the API call in the first place (i.e. describe the bug we were seeing)?

Reply:
You can see some examples of named constants here in this code https://github.com/allenai/scholar-reader/blob/2412d2364be3783166456b9e806a23d4596752a9/ui/src/ScholarReader.tsx#L420-L421

Reply:
How does this work after I combine all three functions into one?

Reply:
Pretty much the same way! You would defined a constant (`e.g., API_PAPER_REQUEST_BATCH_SIZE`) to 50 at the top of the `papers/` function, and refer to that variable when you're splitting up the list of s2Ids.

Reply:
already done that! that's pretty much the only constant in the function?

Reply:
ah I see! Did you push the updates? I don't think I'm seeing that version of the code in the PR, though I might just be looking at the wrong version of it XD

Reply:
I'll push it now!